docs/DevGuide/Protocols.md

```diff
@@ -71,41 +74,45 @@ placed in directory that also has the `greet` function.
 
 # Protocol users: Testing protocol conformance {#protocols_testing_conformance}
 
+We are currently testing protocol conformance as part of our unit tests, so that
+the `tt::conforms_to` metafunction only needs to check if a type inherits off
+the protocol, but doesn't need to check the protocol's implementation. This is
+both to keep compile times low and to have a SFINAE-friendly way of checking
+protocol conformance. Full protocol conformance is tested with the
+`tt::assert_conforms_to` metafunction.
+
 Any class that indicates it conforms to the protocol must test that it actually
-does using the `test_protocol_conformance` metafunction from
-`tests/Unit/ProtocolTestHelpers.hpp`:
+does using the `tt::assert_conforms_to` metafunction:
 
 \snippet Utilities/Test_ProtocolHelpers.cpp test_protocol_conformance
 
 # Protocol authors: Protocols must be unary type traits {#protocols_author}
 
-When you author a new protocol, keep in mind that protocols must be unary type
-traits. This means they take a single template parameter (typically named
-`ConformingType`) and inherit from `std::true_type` or `std::false_type`
-depending on whether the `ConformingType` fullfills the protocol's requirements.
-Make sure to implement the protocol in a SFINAE-friendly way. You may find the
-macros in `Utilities/TypeTraits.hpp` useful. For example, we use
-`CREATE_IS_CALLABLE` in the protocols above for testing the existence and return
-type of a member function.
-
-Occasionally, you might be tempted to add additional template parameters to the
-protocol. In those situations, make the additional parameters part of your
-protocol instead. The reason for this guideline is that protocols
-will always be used as unary type traits when inheriting from
-`tt::ConformsTo<Protocol>`. Therefore, any template parameters of the protocol
-must also be template parameters of their conforming classes, which means the
-protocol can just check them.
-
-For example, we could be tempted to follow this antipattern:
+To author a new protocol you implement a class that provides a `test`
+metafunction and detailed documentation. The `test` metafunction takes a single
+template parameter (typically named `ConformingType`) and checks that it
+conforms to the requirements laid out in the protocol's documentation. Its
+purpose is to provide diagnostic messages as compiler errors to understand why a
+type fails to conform to the protocol. You can use `static_assert`s or trigger
+standard compiler errors where appropriate. See the protocols defined above for
+examples.
+
+Occasionally, you might be tempted to add template parameters to the
+protocol. In those situations, make the parameters part of your
```

Comment:
I find that the first and second sentence in this paragraph seem to directly contradict one another. "You may want to do blah. But instead, you should do blah." Instead, I would've expected it to read "You may want to do blah. Instead, you should do X."